front/pages/api/v1/w/[wId]/data_sources/[dsId]/join-slack.ts

```diff
@@ -0,0 +1,165 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+
+import { getDataSourceViewFromVaultAndDataSourceId } from "@app/lib/api/data_source_view";
+import { getGlobalVault } from "@app/lib/api/vaults";
+import { withPublicAPIAuthentication } from "@app/lib/api/wrappers";
+import type { Authenticator } from "@app/lib/auth";
+import { apiError } from "@app/logger/withlogging";
+
+/**
+ * @swagger
+ * /api/v1/w/{wId}/data_sources/{dsId}/join-slack:
+ *   post:
+ *     summary: Join a Slack channel to a data source
+ *     description: Add a new Slack channel to the list of channels associated with a data source.
+ *     tags:
+ *       - Data Sources
+ *     security:
+ *       - BearerAuth: []
+ *     parameters:
+ *       - in: path
+ *         name: wId
+ *         required: true
+ *         description: Unique string identifier for the workspace
+ *         schema:
+ *           type: string
+ *       - in: path
+ *         name: dsId
+ *         required: true
+ *         description: Unique string identifier for the data source
+ *         schema:
+ *           type: string
+ *     requestBody:
+ *       required: true
+ *       content:
+ *         application/json:
+ *           schema:
+ *             type: object
+ *             required:
+ *               - newChannelId
+ *             properties:
+ *               newChannelId:
+ *                 type: string
+ *                 description: The ID of the new Slack channel to add
+ *     responses:
+ *       200:
+ *         description: Successfully added the new channel
+ *         content:
+ *           application/json:
+ *             schema:
+ *               type: object
+ *               properties:
+ *                 success:
+ *                   type: boolean
+ *                   example: true
+ *                 dataSourceView:
+ *                   $ref: '#/components/schemas/DataSourceView'
+ *       400:
+ *         description: Bad request (invalid input)
+ *       401:
+ *         description: Unauthorized
+ *       404:
+ *         description: Workspace or data source not found
+ *       405:
+ *         description: Method not allowed
+ *       500:
+ *         description: Internal server error
+ */
+async function handler(
```

Comment:
Should we instead create an API that lets you update the data source view? this would be more generic?

Reply:
Is that something we want to be able to do through the public API?

Reply:
Eventually yes. Right now probably only for internal (system key) usage. But it would have a good chance of being reused vs this one cannot and is just debt we're taking :)

Reply:
Alright, will do that then 🤓